packages/bull-vault/test/integration-test/BullStrategyTestFork.t.sol

```diff
@@ -87,17 +90,55 @@ contract BullStrategyTestFork is Test {
 
     function testInitialDeposit() public {
         uint256 crabToDeposit = 10e18;
+        uint256 bullCrabBalanceBefore = bullStrategy.getCrabBalance();
 
         vm.startPrank(user1);
         (uint256 wethToLend, uint256 usdcToBorrow) = _deposit(crabToDeposit);
         vm.stopPrank();
 
+        uint256 bullCrabBalanceAfter = bullStrategy.getCrabBalance();
+
+        assertEq(bullCrabBalanceAfter.sub(crabToDeposit), bullCrabBalanceBefore);
         assertEq(bullStrategy.balanceOf(user1), crabToDeposit);
         assertEq(IEulerDToken(dToken).balanceOf(address(bullStrategy)), usdcToBorrow);
         assertTrue(wethToLend.sub(IEulerEToken(eToken).balanceOfUnderlying(address(bullStrategy))) <= 1);
         assertEq(IERC20(usdc).balanceOf(user1), usdcToBorrow);
     }
 
+    function testSecondDeposit() public {
+        uint256 crabToDepositInitially = 10e18;
+        uint256 bullCrabBalanceBefore = bullStrategy.getCrabBalance();
+
+        vm.startPrank(user1);
+        (uint256 wethToLend, uint256 usdcToBorrow) = _deposit(crabToDepositInitially);
+        vm.stopPrank();
+
+        uint256 bullCrabBalanceAfter = bullStrategy.getCrabBalance();
+
+        assertEq(bullCrabBalanceAfter.sub(crabToDepositInitially), bullCrabBalanceBefore);
+        assertEq(bullStrategy.balanceOf(user1), crabToDepositInitially);
+        assertEq(IEulerDToken(dToken).balanceOf(address(bullStrategy)), usdcToBorrow);
+        assertTrue(wethToLend.sub(IEulerEToken(eToken).balanceOfUnderlying(address(bullStrategy))) <= 1);
+        assertEq(IERC20(usdc).balanceOf(user1), usdcToBorrow);
+
+        bullCrabBalanceBefore = bullStrategy.getCrabBalance();
+        uint256 userUsdcBalanceBefore = IERC20(usdc).balanceOf(user1);
+        uint256 userBullBalanceBefore = bullStrategy.balanceOf(user1);
+        uint256 crabToDepositSecond = 7e18;
```

Comment:
going to fuzz this one?

Reply:
This is @haythem96's deposit PR, not relevant to Leverage Rebalance.